### ISSUE TITLE
NeuralLinear: Gradient Clipping

### ISSUE THREAD

**Issue by philippkolbe** [neural-bandits/calvera#150]
Usually, in pytorch you would set the gradient clip value on the trainer:
```python
    trainer = pl.Trainer(
        max_epochs=1,
        max_steps=1000,
        logger=logger,
        gradient_clip_val=0.5,
        log_every_n_steps=1,
        enable_progress_bar=False,
        enable_model_summary=False,
        enable_checkpointing=False,
    )
```

This works well for NeuralUCB and NeuralTS but fails for NeuralLinear because you can't set 
Therefore, think of a way how to nicely implement gradient_clipping for all 3 algorithms.

One way would be to not turn off automatic_optimization but always return a zero vector in the LinearBandits.
